src/Controller/AdminController.php

```diff
@@ -44,4 +46,66 @@ protected function isActionAllowed($actionName)
 
         return parent::isActionAllowed($actionName);
     }
+
+    /**
+     * Use renderView()
+     *
+     * @param string $actionName   The name of the current action (list, show, new, etc.)
+     * @param string $templatePath The path of the Twig template to render
+     * @param array  $parameters   The parameters passed to the template
+     */
+    protected function renderViewTemplate($actionName, $templatePath, array $parameters = array())
+    {
+        return $this->renderView($templatePath, $parameters);
+    }
+
+    /**
+     * The method that is executed when the user performs a 'new ajax' action on an entity.
+     *
+     * @return JsonResponse
+     */
+    protected function newAjaxAction()
+    {
+        $this->dispatch(EasyAdminEvents::PRE_NEW);
+
+        $entity = $this->executeDynamicMethod('createNew<EntityName>Entity');
+
+        $easyadmin = $this->request->attributes->get('easyadmin');
+        $easyadmin['item'] = $entity;
+        $this->request->attributes->set('easyadmin', $easyadmin);
+
+        $fields = $this->entity['new']['fields'];
+
+        $newForm = $this->executeDynamicMethod('create<EntityName>NewForm', [$entity, $fields]);
+
+        $newForm->handleRequest($this->request);
+        if ($newForm->isSubmitted() && $newForm->isValid()) {
+            $this->dispatch(EasyAdminEvents::PRE_PERSIST, ['entity' => $entity]);
+            $this->executeDynamicMethod('persist<EntityName>Entity', [$entity]);
+            $this->dispatch(EasyAdminEvents::POST_PERSIST, ['entity' => $entity]);
+
+            return new JsonResponse(['option' => ['id' => $entity->getId(), 'text' => (string) $entity]]);
+        }
+
+        $this->dispatch(EasyAdminEvents::POST_NEW, [
+            'entity_fields' => $fields,
+            'form' => $newForm,
+            'entity' => $entity,
+        ]);
+
+        $parameters = [
+            'form' => $newForm->createView(),
+            'entity_fields' => $fields,
+            'entity' => $entity,
+        ];
+
+        if (isset($this->entity['templates']['new_ajax'])) {
+            $templateNewAjax = $this->entity['templates']['new_ajax'];
+        } else {
+            $templateNewAjax = '@EasyAdminExtension/default/new_ajax.html.twig';
+        }
+
+        $template = $this->executeDynamicMethod('renderView<EntityName>Template', array('new_ajax', $templateNewAjax, $parameters));
```

Comment:
Please use default `renderTemplate` method that may be overriden by controller. If you want to implement a dynamic `renderView<EntityName>Template`, it should be requested on original EasyAdmin bundle, not partially added for this unique feature.

Reply:
There is a dynamic method `render<EntityName>Template` so you can use it !

Reply:
renderTemplate return $this->render();
it is necessary:
renderViewTemplate return $this->renderView();
that's why new function

Reply:
Ok, but why do you need it as a dynamic method ? Template is configurable so why defining a new dynamic method here ?

```php
if (isset($this->entity['templates']['new_ajax'])) {
    $templateNewAjax = $this->entity['templates']['new_ajax'];
} else {
    $templateNewAjax = '@EasyAdminExtension/default/new_ajax.html.twig';
}

$html = $this->renderView($templateNewAjax, $parameters);

return new JsonResponse(['html' => $html]);
```

Reply:
ok